### ISSUE TITLE
Is there any way to generate one report per `test_settings`?

### ISSUE THREAD

**Issue by ncochard** [mucsi96/nightwatch-cucumber#256]
Given the following configuration...

	require('nightwatch-cucumber')({
		cucumberArgs: [
			'--format', 'pretty',
			'--format', 'json:reports/cucumber.json',
			'features'
		]
	});
	module.exports = {
		...
		test_workers: true,//Tests running in parallel
		...
		test_settings: {
			default: {
				launch_url: 'http://ondemand.saucelabs.com:80',
				...
			},
			chrome: {
				...
			},
			firefox: {
				...
			},
			ie11: {
				...
			}
	    }
	};

And given the following command...

	nightwatch --env chrome,firefox,ie11

All my tests are executed in parallel and the reports for all these tests are merged into one file `cucumber.json`.

The problem is that I cannot distinguish in the report what test was executed for _firefox_ or _chrome_ or _IE_.

Is there any way to generate one report per platform?

**Comment by danielgecko** [comment]
Hey,

I believe there is

```
test_setting: {
default: {
output_folder: 'report/'
}
}
```
You could generate a separate report folder for each configuration and then spin up a report for each one

**Comment by ncochard** [comment]
Thank you for the suggestion... I tried to add the following config, but it does not help. I only get one report `reports/cucumber.json` with no way of distinguishing between the result from *chome*, *firefox* or *IE*.

	require('nightwatch-cucumber')({
		cucumberArgs: [
			'--format', 'pretty',
			'--format', 'json:reports/cucumber.json',
			'features'
		]
	});
	module.exports = {
		...
		test_workers: true,//Tests running in parallel
		...
		test_settings: {
			default: {
				launch_url: 'http://ondemand.saucelabs.com:80',
				...
			},
			chrome: {
				output_folder: 'reports/chrome', // <-- does not help!
				...
			},
			firefox: {
				output_folder: 'reports/firefox', // <-- does not help!
				...
			},
			ie11: {
				output_folder: 'reports/ie11', // <-- does not help!
				...
			}
	    }
	};

**Comment by danielgecko** [comment]
Hmm I had a look into it and can only get the output_folder option working from the top level rather than in each profile.

I looked for a solution and kind of found one than is not exactly ideal but could possibly work:

https://github.com/cucumber/cucumber-js/blob/master/docs/cli.md#profiles

You could set up multiple profiles with different cucumberArgs for the --format', 'json:reports/cucumber.json' configuration so that it saved to different locations and then when you run your script you will just have to specify the profile to use.

There's maybe a nicer way to do this....

**Comment by danielgecko** [comment]
I also found a related issue:

https://github.com/mucsi96/nightwatch-cucumber/issues/237

which has a solution that could be changed slightly to work for you.

**Comment by ncochard** [comment]
Hello

Thank you @danielgecko for looking into this.

Yes, you are correct. #237 will solve my issue. Feel free to close my issue (#256) and I'll follow the discussion on #237.

I was actually considering a solution similar to what was explained in that thread. (See below.)

	const argv = require('yargs-parser')(process.argv.slice(2));
	const { env } = argv;
	require('nightwatch-cucumber')({
		cucumberArgs: [
			...
			'--format', `json:reports/cucumber-${env}.json`,
			...
		]
	});

I'm considering writing a wrapper for nightwatch that will [spawn](https://www.npmjs.com/package/cross-spawn) the following commands (in parallel) `nightwatch --env ie11`, `nightwatch --env firefox`, `nightwatch --env chrome`. This would generate the following reports: `cucumber-ie11.json`, `cucumber-firefox.json` and `cucumber-chrome.json`. And my wrapper will gather all these reports into one folder for easy browsing. I'll share my code when it's complete!

Best regards
